src/SimpleGraphs/generators/staticgraphs.jl

```diff
@@ -536,6 +536,54 @@ function double_binary_tree(k::Integer)
     return g
 end
 
+"""
+    regular_tree(k::Integer, z::integer)
+
+Create a [regular tree](https://en.wikipedia.org/wiki/Bethe_lattice), 
```

Comment:
That wikipedia article might be a bit confusing, because it talks about infinite trees in the introduction.

Maybe we can decribe that a bit more clearly here?

If I understood correctly, what you want to create is a perfect k-ary tree such as described here? https://en.wikipedia.org/wiki/M-ary_tree#Types_of_m-ary_trees